### PR TITLE
Added HorizontalPodAutoScaler resource handler

### PR DESCRIPTION
**Pull request by SameeraGrandhi**

### CODE REVIEW

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/predicates/ManifestPredicates.java [hunk 1]
```diff
@@ -22,160 +22,175 @@
...
+
+    public static Predicate<ServiceSpec> haveConfigmapVolume() {
+        return serviceSpec -> {
+            String propsVolumePath = null;
+            try {
+                propsVolumePath = serviceSpec.get(HyscaleSpecFields.propsVolumePath, String.class);
+            } catch (HyscaleException e) {
+                return false;
+            }
+            if (!StringUtils.isBlank(propsVolumePath)) {
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/predicates/ManifestPredicates.java]: Change this to isNotBlank

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/predicates/ManifestPredicates.java [hunk 2]
```diff
@@ -22,160 +22,175 @@
...
-	}
+    public static Predicate<ServiceSpec> haveSecretsVolume() {
+        return serviceSpec -> {
+            String secretsVolumePath = null;
+            try {
+                secretsVolumePath = serviceSpec.get(HyscaleSpecFields.secretsVolumePath, String.class);
+            } catch (HyscaleException e) {
+                return false;
+            }
+            if (!StringUtils.isBlank(secretsVolumePath)) {
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/predicates/ManifestPredicates.java]: Change this to isNotBlanl

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/provider/PluginTemplateProvider.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        for (PluginTemplateType each : PluginTemplateType.values()) {
+            ConfigTemplate configTemplate = new ConfigTemplate();
+            configTemplate.setRootPath(TEMPLATES_PATH);
+            configTemplate.setTemplateName(each.getTemplateFile());
+            templateMap.put(each.getTemplateFile(), configTemplate);
+        }
+    }
+
+    public ConfigTemplate get(PluginTemplateType type) {
+        if (type == null) {
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/provider/PluginTemplateProvider.java]: We can move the null check for template map here and later just return templateMap.get(type.getTemplateFile())

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/predicates/ManifestPredicates.java [hunk 4]
```diff
@@ -22,160 +22,175 @@
...
-			if(secrets == null){
-			    return false;
-            }
-			if(secrets.getType() == SecretType.MAP){
-			    MapBasedSecrets mapBasedSecrets = (MapBasedSecrets) secrets;
-			    if(mapBasedSecrets != null && !mapBasedSecrets.isEmpty()){
-			        return true;
+    public static Predicate<ServiceSpec> getSecretsPredicate() {
+        return serviceSpec -> {
+            //	Secrets secrets = null;
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/predicates/ManifestPredicates.java]: Remove the comment

File: manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/AutoScalingPluginHandler.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        ManifestSnippet snippet = new ManifestSnippet();
+        snippet.setKind(ManifestResource.HORIZONTAL_POD_AUTOSCALER.getKind());
+        snippet.setPath("spec");
+        snippet.setSnippet(yamlString);
+        List<ManifestSnippet> snippetList = new LinkedList<>();
+        snippetList.add(snippet);
+        return snippetList;
+    }
+
+    //TODO min<max && min>0 check
```
Comment (tushar-tushar) [inline manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/AutoScalingPluginHandler.java]: This we are already checking right?

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1HorizontalPodAutoScalerHandler.java [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+        } catch (HyscaleException ex) {
+            logger.debug("Error while getting HorizontalPodAutoScaler  {} in namespace {} for Update, creating new", name, namespace);
+            V1HorizontalPodAutoscaler horizontalPodAutoscaler = create(apiClient, resource, namespace);
+            return horizontalPodAutoscaler != null ? true : false;
+        }
+        WorkflowLogger.startActivity(DeployerActivity.DEPLOYING_HORIZONTAL_AUTO_SCALER);
+        try {
+            String resourceVersion = existingHorizontalPodAutoscaler.getMetadata().getResourceVersion();
+            resource.getMetadata().setResourceVersion(resourceVersion);
+            autoscalingV1Api.replaceNamespacedHorizontalPodAutoscaler(name, namespace, existingHorizontalPodAutoscaler, TRUE, null);
```
Comment (harijvs6) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1HorizontalPodAutoScalerHandler.java]: resource?

**Pull request merged**